### PR TITLE
My Plan: Add Checklist CTA

### PR DESCRIPTION
**Pull request by sirreal**
Add a checklist CTA to the My Plan header based on https://github.com/Automattic/jetpack/issues/11997#issuecomment-494155147

This introduces _yet another primary button_ to this view. https://github.com/Automattic/jetpack/pull/12430 removes the primary buttons from all of the feature cards in the view.

#### Screens

<details>

<summary>Expand for screens</summary>

placeholder
free-mob
free-desk
personal-mob
personal-desk
premium-mob
premium-desk
pro-mob
pro-desk

</details>

#### Testing instructions:

* From `/wp-admin/admin.php?page=jetpack#/my-plan` on a Jetpack site
* You should see the checklist CTA button like in the above screens.
* For sites with different plans, does the button appear correctly? Does the link work as expected?

#### Proposed changelog entry for your changes:

none

### CODE REVIEW

**Comment by jeherve** [comment]
Should we show the button on free plans as well? It seems odd, since there isn't much to set up at that point, you are just invited to purchase a plan?

**Comment by simison** [comment]
>Should we show the button on free plans as well? It seems odd, since there isn't much to set up at that point, you are just invited to purchase a plan?

There will be some new stuff coming to the checklist sometime after 7.4 release for free sites as well; lazy loading images & CDN for example.

**Comment by jeffgolenski** [comment]
We definitely want to display this for free. As @simison mentioned, more things are coming to the checklist soon!

@sirreal I just testing this - I'm noticing a difference in styles. For paid plans the button is in the correct place, but but free it is not. 

Looks great on mobile. Can you just bump the `p` text size to 14px?

Screen Shot 2019-05-21 at 10 47 03 AM
Screen Shot 2019-05-21 at 10 44 57 AM

**Comment by simison** [comment]
Should we address multiple primary color buttons on the same page as a follow up to this?

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (simison) [inline _inc/client/my-plan/my-plan-header/checklist-cta.js]: is `placeholder` prop used anywhere?

Comment (simison) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: Looks like usage of it was removed in commit 27af4f6e5. Let's clean up here, too, if it's un-used.

**Review by simison — DISMISSED** [review]
Code is looking good :thumbsup:

Just small `placeholder` change I'd like to see before merging.

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (sirreal) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: 👍 cf369ad324dcfdb1a56905e36f1a9a6c5f95bd0e

**Review by simison — DISMISSED** [review]

File: _inc/client/my-plan/my-plan-header/index.js
Comment (sirreal) [inline _inc/client/my-plan/my-plan-header/index.js]: This one class was inconsistent with the rest.

**Comment by sirreal** [comment]
Rebased so the primary CTAs are no longer repeated (#12430).

> I just testing this - I'm noticing a difference in styles. For paid plans the button is in the correct place, but but free it is not.

There was a misnamed class in exactly 1 place, which was the class I'd based the changes on. I fixed this class name so the different plans should be consistent.

This also highlighted a lot of what appear to be leftover classes from previous plans pages. I've removed many rules that appear to be unused after searching for references.

> Looks great on mobile. Can you just bump the `p` text size to 14px?

I've taken care of that 👍 

<details>

<summary>Here are updated mobile/desktop screenshots for all plans</summary>

placeholder
free-mob
free-desk
personal-mob
personal-desk
premium-mob
premium-desk
pro-mob
pro-desk


</details>

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (ockham) [inline _inc/client/my-plan/my-plan-header/checklist-cta.js]: Is the `__` alias going to work fine here? :thinking: It will probably be handled by the plugin repo's string extraction handling, which is made for `@wordpress/i18n`, and the string will end up in the translation files generated by the plugin repo. But `i18n-calypso` won't load those, will it?

**Review by ockham — COMMENTED** [review]
One concern [here](https://github.com/Automattic/jetpack/pull/12429/files#r286427482). Appears to be working well otherwise! I also couldn't find any other uses of those stale class names.

Feel free to ship after addressing the i18n question!

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (sirreal) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: This is nothing new or novel, it's how things are done around here 🙂 

https://github.com/Automattic/jetpack/blob/510a46bda82d61245e10f392933a09d839218871/_inc/client/my-plan/my-plan-body.jsx#L8
https://github.com/Automattic/jetpack/blob/510a46bda82d61245e10f392933a09d839218871/_inc/client/my-plan/my-plan-body.jsx#L129

Comment (simison) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: I think there's a script somewhere which pulls these out to a php file, and we're not relying on wp.org process to extract these. 🤔 @zinigor might know better.

Comment (jeherve) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: The magic happens here:
https://github.com/Automattic/jetpack/blob/7b6c7495020716d3d89fd7b0b8d0f4f3093348bb/gulpfile.babel.js#L204

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (jeherve) [inline _inc/client/my-plan/my-plan-header/checklist-cta.js]: Do we want to add tracking to clicks on that button?

**Review by jeherve — APPROVED** [review]
This looks good, and tests well for me. I think this should be good to merge after a design review.

I mentioned the lack of tracking on that button below, but I don't think that's blocking here, given that we are missing tracking on other buttons in there. It could be added in a future PR I think.

**Comment by jeffgolenski** [comment]
>  mentioned the lack of tracking on that button below, but I don't think that's blocking here, given that we are missing tracking on other buttons in there. It could be added in a future PR I think.

Thanks for bringing this up. Yes, we need be adding tracking to all new buttons that are added, every single time. If we don't know if / how our customers are using what we design and build then we're just wasting our time. Can we get a tracks event in there @sirreal? Thank ya! 

cc  @jeherve @simison @ockham

**Comment by sirreal** [comment]
> Can we get a tracks event in there @sirreal? 

@jeffgolenski yes, but I'd prefer to add it in a follow up since this has the approvals already. Can we consider this design approved?

**Review by jeffgolenski — APPROVED** [review]
Did another run through. Looks great!

File: _inc/client/my-plan/my-plan-header/checklist-cta.js
Comment (ockham) [reply _inc/client/my-plan/my-plan-header/checklist-cta.js]: TIL

**Comment by sirreal** [comment]
Tracks here: https://github.com/Automattic/jetpack/pull/12449

**Pull request merged**
